tokio-trace/tokio-trace-core/src/dispatcher.rs

```diff
@@ -63,6 +73,30 @@ pub fn with_default<T>(dispatcher: &Dispatch, f: impl FnOnce() -> T) -> T {
     let _guard = State::set_default(dispatcher.clone());
     f()
 }
+
+/// Sets this dispatch as the global default for the duration of the entire program.
+/// Will be used as a fallback if no thread-local dispatch has been set in a thread
+/// (using `with_default`.)
+///
+/// Can only be set once; subsequent attempts to set the global default will fail.
+/// Returns Err if the global default has already been set.
```

Comment:
I think it's worth cautioning that libraries should generally not call this function, to avoid conflicting with a global subscriber set by their dependents.